pkg/cgroups/cgroups_linux.go

```diff
@@ -58,9 +58,7 @@ func cgrpCheckOrMountLocation(cgroupRoot string) error {
 
 	// If the custom location has no mount, let's mount there.
 	if !mounted {
-		if err := mountCgroup(); err != nil {
-			return err
-		}
+		return mountCgroup()
```

Comment:
Why do we return even in the case of `err == nil`?

Reply:
It is to avoid checking `cgroupInstance`. When the cgroup is not yet mounted, this variable receives `false` as return of `mountinfo.IsMountFS`. Because of that, the function returns the error:

```C
if !cgroupInstance {
    return fmt.Errorf("Mount in the custom directory %s has a different filesystem than cgroup2", cgroupRoot)
}
```
As a consequence, the function `func getCgroupRootForKind() (string, error)` is never called. This is not a critical bug, but it causes cilium to use a cgroupv2 sub-hierarchy that is subtly different than expected.

Reply:
Thanks for the explanation. I was not aware that not much happens in the function after we mount.